parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/BaseWorkFlowTask.java

```diff
@@ -116,6 +117,21 @@ public String getRequiredParameterValue(String parameterName) throws MissingPara
 				});
 	}
 
+	public boolean validateWorkflowParameters(WorkContext workContext) {
```

Comment:
We need something like this one at execution level.
But that might be tricky if some of the arguments are filled during the execution (such in the case of m2k by a questionnaire) so maybe it is good enough to let the task developer set it.

Last thought about parameters validity: perhaps we can add annotation to a task such as @RequiredAllArguments and invoke this method in the aspect before invoking the task execution.
Just a thought :-) and entirely out of the scope of this PR.

Reply:
it would more be @EnforceRequiredArguments as the optional args are ignored and yes, the task is the lowest level; it's the one which need the actual arguments so it's better to check it on the task level as well and as you mentioned,  this allows the settings of the arguments at any level prior to the task 

I created a ticket: https://issues.redhat.com/browse/FLPATH-485